src/lib.rs

```diff
@@ -160,6 +161,12 @@ impl Client {
         self
     }
 
+    pub fn with_param(self, name: &str, value: impl Serialize) -> Result<Self, String> {
```

Comment:
What's the purpose of `Client::with_param`? Parameters are specific for queries, not the whole client

Reply:
Generally, but if someone wants to have a set of queries which share a parameter they can put it on the client

For example, in a multi tenant app, tenantid could be put as a parameter in the function creating client

Reply:
What's the obscure outcomes you're thinking of? Better people use this than roll it themselves with options

Reply:
+1 to @slvrtrn, let's remove `Client::with_param` for now until someone asks it explicitly.

Reply:
... and finally, merge it; this is the last question.

Reply:
removed